### PR TITLE
Only updated edited media when exiting edit-mode

### LINKED ISSUE #1945: Media Zooming: Group actions in history and remove reset button
## Bug Description

Currently, when zooming Media, each zoom step is considered as a separate history entry.

## Expected Behaviour

Zooming should be considered as one history entry.

## Steps to Reproduce

1. Add Media
2. Go to edit mode and change the zoom setting for a few times from the slider.
3. Now exit the edit mode and use History Undo.
4. See all the Zoom steps being shown one by one as opposed to all together.

## Screenshots

zooming

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by barklund**
## Summary

This removes the immediate update of the data storage of the media item when edited inside the media editor. Instead, edited properties are kept in a local state used for display inside the editor only, and the global store is updated only when the editor is unmounted.

This also removes the reset button inside the media editor, as it was confusing in terms of what it actually resat.

~However, if <kbd>esc</kbd> is pressed, the media editor edits are discarded.~ Not relevant anymore.

## Relevant Technical Choices

* Didn't have to use (and modify) our transform handler at all, as the state can be kept purely internal to the media edit component!

## To-do

* [x] Remove reset button
* [x] Keep edits internal
* [x] Update global store only on unmount
* [x] ~Don't update global store on unmount due to `esc`~ This has been reverted, ignore
* [x] Update and amend tests as required

## User-facing changes

* All edits in media edit-mode is now confined to a single history-entry and can be undone by pressing undo just once

## Testing Instructions

* Add a media element
* Double-click to enter edit mode
* Make multiple separate edits of pan and/or zoom
* Click outside the element or press <kbd>esc</kbd> to exit edit-mode 
* Observe that the media has been updated as desired
* Press undo
* Observe that all the edits are undone in a single undo-step

---

Fixes #1945

### CODE REVIEW

File: assets/src/edit-story/elements/media/edit.js [hunk 1]
```diff
@@ -107,25 +109,62 @@ function MediaEdit({ element, box }) {
   const [croppedMedia, setCroppedMedia] = useState(null);
   const [cropBox, setCropBox] = useState(null);
   const elementRef = useRef();
+  const scaleRef = useRef(null);
```
Comment (barklund) [inline assets/src/edit-story/elements/media/edit.js]: Default null is required here to make `useKeyDownEffect` work correctly when the element hasn't mounted yet.

File: assets/src/edit-story/elements/media/edit.js [hunk 2]
```diff
@@ -107,25 +109,62 @@ function MediaEdit({ element, box }) {
...
+
+  // Actually update the properties of the current element on unmount
   const { updateElementById } = useStory((state) => ({
     updateElementById: state.actions.updateElementById,
   }));
-  const setProperties = useCallback(
-    (properties) => updateElementById({ elementId: id, properties }),
-    [id, updateElementById]
-  );
+  const updateActualProperties = useCallback(() => {
```
Comment (miina) [inline assets/src/edit-story/elements/media/edit.js]: The naming `actualProperties` is a bit confusing to read -- I assumed at first that `actualProperties` would be the current actual properties that are applied to the media, however, the actual properties are the globally saved ones, right?

Could we perhaps name the local properties to `localProperties` and keep the global properties named just `properties` (and `setProperties`)? WDYT?

**Review by miina — COMMENTED** [review]
Awesome that we can do this just by storing the properties locally.

LGTM other than the Esc logic -- I think it should probably match the logic of the Text element for consistency (exit the edit mode but keep the changes).

File: assets/src/edit-story/elements/media/edit.js (hunk 2 above)
Comment (barklund) [reply assets/src/edit-story/elements/media/edit.js]: Fixed in 3187660cc.

**Review by miina — APPROVED** [review]

File: assets/src/edit-story/elements/media/edit.js [hunk 3]
```diff
@@ -107,25 +108,47 @@ function MediaEdit({ element, box }) {
   const [croppedMedia, setCroppedMedia] = useState(null);
   const [cropBox, setCropBox] = useState(null);
   const elementRef = useRef();
+  const [localProperties, setLocalProperties] = useState({});
+  const lastLocalProperties = useRef({});
```
Comment (miina) [inline assets/src/edit-story/elements/media/edit.js:115]: Couldn't we use just `localProperties` and skip the `lastLocalProperties`?

Comment (barklund) [reply assets/src/edit-story/elements/media/edit.js:115]: We need it in a ref to have a stable function to pass to the unmount-function. But we also need it in a state, to make the component re-render on updates. So, we need both here.

**Comment by samitron7** [comment]
Left the same comment in #[1945](https://github.com/google/web-stories-wp/issues/1945):

"Zooming looks to be grouped into one history entry but there is a regression with cropping (filed ticket 5464)"

**Pull request merged**